torch/csrc/autograd/python_variable.cpp

```diff
@@ -1519,18 +1520,28 @@ void concrete_dispatch_fn(const c10::impl::PyInterpreter*, const c10::OperatorHa
     if (default_ivalue != ivalue) {
       break;
     }
-    default_suffix_len++;
   }
 
-  auto args = py::reinterpret_steal<py::object>(PyTuple_New(num_arguments - default_suffix_len));
+  // Find the split point between kwarg-only and regular.
+  // (Why +1?  As in the normal indexing convention, end is exclusive, begin is
+  // inclusive.  So if idx is actually kwarg only--we don't know yet--it is
+  // only included if I set end to +1)
+  const int64_t kwarg_only_end = idx + 1;
+  for (; idx >= 0; idx--) {
+    const auto& arg = schema.arguments()[idx];
+    if (!arg.kwarg_only()) {
+      break;
+    }
+  }
+  const int64_t kwarg_only_begin = idx + 1;
+  const int64_t positional_end = idx + 1;
 
-  // TODO: actually populate kwargs sometimes?  At the moment, every argument
-  // just gets passed positionally
+  auto args = py::reinterpret_steal<py::object>(PyTuple_New(positional_end));
   py::dict kwargs;
 
-  for (int64_t idx = 0; idx < arguments.size() - default_suffix_len; idx++) {
-    auto& ivalue = arguments[idx];
-    // Search for Tensors (as they may have the torch functions we need)
+  // Find overloaded tensors
+  for (int64_t idx = 0; idx < kwarg_only_end; idx++) {
```

Comment:
nit: idx variable re-use for a very different meaning compared to the loops above